x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,193 @@
+
+export const INDEX_TIMEOUT_IN_HOURS = 5;
+
+const isFindingsExists = async (esClient: ElasticsearchClient): Promise<boolean> => {
+  try {
+    const queryResult = await esClient.search({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+      query: {
+        match_all: {},
+      },
+      size: 1,
+    });
+
+    const hasLatestFinding = !!queryResult.hits.hits.length;
+
+    return hasLatestFinding ? true : false;
+  } catch (e) {
+    return false;
+  }
+};
+
+const getHealthyAgents = async (
+  soClient: SavedObjectsClientContract,
+  cspPackagePolicies: ListResult<PackagePolicy>,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<number> => {
+  const agentPolicies = await getCspAgentPolicies(
+    soClient,
+    cspPackagePolicies.items,
+    agentPolicyService
+  );
+  const enrichAgentPolicies = await addRunningAgentToAgentPolicy(agentService, agentPolicies);
+  const initialValue = 0;
+  const totalAgents = enrichAgentPolicies
+    .map((agentPolicy) => (agentPolicy.agents ? agentPolicy.agents : 0))
+    .reduce((previousValue, currentValue) => previousValue + currentValue, initialValue);
+
+  return totalAgents;
+};
+
+const getTimeDeltaSinceInstallation = (packageInfo: Installation): number => {
+  const installTime = packageInfo!.install_started_at;
+
+  // calculate time delta and convert to hours
+  const dt = (new Date().getTime() - new Date(installTime).getTime()) / (1000 * 60 * 60);
+
+  return dt;
+};
+
+const getStatus = async (
+  esClient: ElasticsearchClient,
+  installedIntegrations: number,
+  healthyAgents: number,
+  timeDeltaSinceInstallation: number
+): Promise<Status> => {
+  if (await isFindingsExists(esClient)) return 'indexed';
+
+  if (installedIntegrations === 0) return 'not installed';
+
+  if (healthyAgents === 0) return 'not deployed';
+
+  if (timeDeltaSinceInstallation < INDEX_TIMEOUT_IN_HOURS) return 'indexing';
+
+  return 'index timeout';
+};
+
+const getCspSetupStatus = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  packageService: PackageService,
+  packagePolicyService: PackagePolicyServiceInterface,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<CspSetupStatus> => {
+  const packageInfo = await packageService.asInternalUser.getInstallation(
+    CLOUD_SECURITY_POSTURE_PACKAGE_NAME
+  );
+
+  const cspPackageInstalled = await getCspPackagePolicies(
+    soClient,
+    packagePolicyService,
+    CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+    { per_page: 10000 }
+  );
+
+  const healthyAgents = await getHealthyAgents(
+    soClient,
+    cspPackageInstalled,
+    agentPolicyService,
+    agentService
+  );
+
+  const latestPkgVersion = (
+    await packageService.asInternalUser.fetchFindLatestPackage(CLOUD_SECURITY_POSTURE_PACKAGE_NAME)
+  )?.version;
+
+  const timeDeltaSinceInstallation = packageInfo
+    ? await getTimeDeltaSinceInstallation(packageInfo)
+    : 0;
```

Comment:
later on we compare this value with another constant which is in `hours` 

i think we should indicate the unit of measure (hours) in this name too